pkg/builder/builder.go

```diff
@@ -420,13 +445,21 @@ func buildVolumes(vdb *vapi.VerticaDB) []corev1.Volume {
 	return vols
 }
 
-// buildScrutinizeVolume returns an emptyDir volume for scrutinize if no volume was
-// specified in the vscr CR. Otherwise the specified volume is returned
-func buildScrutinizeVolume(vscr *v1beta1.VerticaScrutinize) corev1.Volume {
+// buildScrutinizeVolumes returns volumes that will be used by the scrutinize pod
+func buildScrutinizeVolumes(vscr *v1beta1.VerticaScrutinize, vdb *vapi.VerticaDB) []corev1.Volume {
+	vols := []corev1.Volume{}
+	if vmeta.UseVClusterOps(vdb.Annotations) &&
+		vmeta.UseNMACertsMount(vdb.Annotations) &&
+		vdb.Spec.NMATLSSecret != "" &&
+		secrets.IsK8sSecret(vdb.Spec.NMATLSSecret) {
+		vols = append(vols, buildNMACertsSecretVolume(vdb))
+	}
 	if vscr.Spec.Volume == nil {
-		return buildDefaultScrutinizeVolume()
+		vols = append(vols, buildDefaultScrutinizeVolume())
+		return vols
 	}
-	return *vscr.Spec.Volume
+	vols = append(vols, *vscr.Spec.Volume)
```

Comment:
If someone provides their own volume, how do we ensure the volume mount will use it? I think its required that any custom volume must be called `scrutinize`

Reply:
There is a function `getScrutinizeVolumeMountName()` that returns the name of the custom volume if any or "scrutinize".